### PR TITLE
Twitter analytics popular tweets

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
Provide refresh functionality for popular tweets widget

### CODE REVIEW

File: components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx [hunk 1]
```diff
@@ -14,62 +14,149 @@
...
         super(props);
 
         this.state = {
             tweetData: [],
+            unreadTweets: [],
             width: this.props.glContainer.width,
             height: this.props.glContainer.height
         };
 
         this.handleResize = this.handleResize.bind(this);
```
Comment (Niveathika92) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx]: Use inline function for resizing,
https://github.com/wso2/analytics-solutions/blob/master/components/sp-solutions/org.wso2.carbon.sp.solutions.http.analytics/widgets/HTTPAnalyticsLatencyComparison/src/HTTPAnalyticsLatencyComparison.jsx#L86

Comment (ruwiniwj) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx]: Done

File: components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx [hunk 2]
```diff
@@ -14,62 +14,147 @@
  * limitations under the License.
  */
 
-import React, {Component} from 'react';
+import React from 'react';
 import Widget from '@wso2-dashboards/widget';
+import {MuiThemeProvider} from 'material-ui/styles/index';
+import {IconButton} from "material-ui";
```
Comment (TanyaM) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx]: Named imports

Comment (ruwiniwj) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx]: Fixed with https://github.com/wso2/analytics-solutions/pull/303

File: components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx [hunk 3]
```diff
@@ -14,62 +14,147 @@
...
+            let numberOfRemovableTweets = result.length - MAX_UNREAD_TWEET_COUNT;
+            result.splice(MAX_UNREAD_TWEET_COUNT, numberOfRemovableTweets < 0 ? 0 : numberOfRemovableTweets);
+
+            this.setState({
+                unreadTweets: result
+            });
+        }
+    }
+
+
```
Comment (dinusha92) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx]: Shall we remove this additional lines?

Comment (ruwiniwj) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.twitter.analytics/widgets/PopularTweets/src/PopularTweets.jsx]: Fixed with https://github.com/wso2/analytics-solutions/pull/306

**Pull request merged**